### PR TITLE
submodule discount and statically link to it

### PR DESCRIPTION
**Pull request by visnup**
This is mostly a work in progress. It does work as-is, but being that I don't know the waf api very well, it's pretty gross. Could use some feedback on it.

The idea is that anywhere Orc/discount builds, this should build w/o a dependency.

Should probably try for a shared/existing dependency first, then fallback to statically linking the submodule'd discount library in.

### CODE REVIEW

File: discount [hunk 1]
```diff
@@ -0,0 +1 @@
+Subproject commit b9c84032eb7b61891c4f239260b33f21537ca1a0
```
Comment (tj) [inline discount:1]: on an npm install this will fail without grabbing the submod first wont it?

Comment (visnup) [inline discount:1]: npm publish pushes whatever submodule was checked out with the rest of the package.

so, if you were to clone then publish, then yes. if you were to clone, submodule init, then publish, then it just works. 

again, all more hacky than I want.

Comment (tj) [inline discount:1]: right right. hmm.. i'll give it a try

Comment (visnup) [inline discount:1]: I have it published as `discount-static` currently. (thought I had unpublished that, but I guess it's still there)

Comment (tj) [inline discount:1]: cool man, seems to be fine i'll merge

**Comment by visnup** [comment]
re-iterating what I want to accomplish with this pull request:

pros:
- no dependency on libmarkdown being pre-installed
  - will just work for heroku, nodejitsu, other PaaS offerings probably
- simpler, more straightforward install (developer has a C compiler anyway, so why not compile discount too?)
- ability to vendor a known working version of discount and bring in upstream as wanted

cons:
- supporting a more complicated preinstall step
  - if libmarkdown doesn't compile, that's not exactly our fault
- blocking a user from using a specific version of discount that they might want

so, it's possible to try to use /usr/lib/libmarkdown or /usr/local/lib/libmarkdown first then fallback to the packaged version, but that's possibly even more complicated for people to understand.

**Pull request merged**